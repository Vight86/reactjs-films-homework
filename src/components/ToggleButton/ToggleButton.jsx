import React from 'react';
import propTypes from 'prop-types';
import styles from './ToggleButton.scss';

const ToggleButton = ({ className }) => (
  <button
    type="button"
    className={styles[className]}
  >
    <span className={styles.toggleGridButtonMask} />
  </button>

);

ToggleButton.propTypes = {
  className: propTypes.string,
};

ToggleButton.defaultProps = {
  className: 'toggleGridButton',
};

export default ToggleButton;
