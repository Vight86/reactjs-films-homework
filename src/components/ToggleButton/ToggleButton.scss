@import '../../utils/variables';

.toggleGridButton {
  position: relative;
  width: 86px;
  height: 32px;
  flex-shrink: 0;
  padding: 0;
  border: none;
  background: transparentize($color: $text-tertiary, $amount: 0.85);
  cursor: pointer;
  outline: none;

  &::before,
  &::after {
    position: absolute;
    z-index: -1;
    left: 50%;
    display: block;
    width: 50%;
    height: 50%;
    background: transparentize($color: $color-secondary, $amount: 0.2);
    content: '';
    transition: transform 0.3s;
  }

  &::before {
    top: 0;
  }

  &::after {
    bottom: 0;
    transition-delay: 0.15s;
  }
}

.active {
  &::before,
  &::after {
    transform: translateX(-100%);
  }
}

.toggleGridButtonMask {
  display: block;
  width: 100%;
  height: 100%;
  background-image: url('./toggleGridButtonMask.svg');
}
