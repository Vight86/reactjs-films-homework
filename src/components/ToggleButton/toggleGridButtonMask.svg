<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Слой_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 86 32" style="enable-background:new 0 0 86 32;" xml:space="preserve">
<style type="text/css">
	.st0{fill-rule:evenodd;clip-rule:evenodd;fill:#F5F6F6;}
</style>
<g>
	<path class="st0" d="M84,0c1.1,0,2,0.9,2,2V0H84z"/>
	<path class="st0" d="M0,30v2h2C0.9,32,0,31.1,0,30z"/>
	<path class="st0" d="M2,0H0v2C0,0.9,0.9,0,2,0z"/>
	<path class="st0" d="M84,13.7H53c-1.1,0-2-0.9-2-2V2c0-1.1,0.9-2,2-2H33c1.1,0,2,0.9,2,2v9.7c0,1.1-0.9,2-2,2H21.1
		c-1.1,0-2-0.9-2-2V2c0-1.1,0.9-2,2-2h-7.2c1.1,0,2,0.9,2,2v9.7c0,1.1-0.9,2-2,2H2c-1.1,0-2-0.9-2-2v8.7c0-1.1,0.9-2,2-2h11.9
		c1.1,0,2,0.9,2,2V30c0,1.1-0.9,2-2,2h7.2c-1.1,0-2-0.9-2-2v-9.7c0-1.1,0.9-2,2-2H33c1.1,0,2,0.9,2,2V30c0,1.1-0.9,2-2,2h20
		c-1.1,0-2-0.9-2-2v-9.7c0-1.1,0.9-2,2-2h31c1.1,0,2,0.9,2,2v-8.7C86,12.8,85.1,13.7,84,13.7z"/>
	<path class="st0" d="M84,32h2v-2C86,31.1,85.1,32,84,32z"/>
</g>
</svg>
