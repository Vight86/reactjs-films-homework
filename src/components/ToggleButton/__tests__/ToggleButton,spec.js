import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ToggleButton from '../index';

const setUp = (className) => {
  const renderer = new ShallowRenderer();
  renderer.render(<ToggleButton className={className} />);
  return renderer.getRenderOutput();
};

describe('render toggleGridButton component', () => {
  it('render component without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });

  it('render component with props', () => {
    const result = setUp('testClass');
    expect(result).toMatchSnapshot();
  });
});
