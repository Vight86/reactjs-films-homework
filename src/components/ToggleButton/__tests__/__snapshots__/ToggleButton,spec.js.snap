// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render toggleGridButton component render component with props 1`] = `
<button
  className="testClass"
  type="button"
>
  <span
    className="toggleGridButtonMask"
  />
</button>
`;

exports[`render toggleGridButton component render component without props 1`] = `
<button
  className="toggleGridButton"
  type="button"
>
  <span
    className="toggleGridButtonMask"
  />
</button>
`;
